spec/Naming.md

```diff
@@ -108,13 +108,52 @@ Identifier :
  * Unique name: {path}
    * URI =   hdfs://{namenode host}:{namenode port}{path}
 
-## Schedulers
-### Airflow
+## Jobs
+### Context
 
-Naming hierarchy:
- * job => workspace/DAG
- * Task => unique within the job
+A `Job` is a recurring data transformation with Inputs and outputs. Each instance is captured as a `Run` with corresponding metadata.
+A `Run` event identifies the `Job` it is an instance of by providing the job’s unique identifier.
+The `Job` identifier is composed of a `Namespace` and a `Name`. The job name is unique within that namespace.
 
-Identifier :
- * Namespace: {scheduler namespace}
- * Unique name: {job}.{task}
+The core property we want to identify about a `Job` is how it changes over time. Different schedules of the same logic applied to different datasets (possibly with different parameters) are different jobs. The notion of a `job` is tied to a recurring schedule with specific inputs and outputs. It could be an incremental update or a full reprocess or even a streaming job.
+ 
+If the same code artifact (for example a spark jar or a templated SQL query) is used in the context of different schedules with different input or outputs then they are different jobs.
+We are interested first in how they affect the datasets they produce.
+
+### Job Namespace and constructing job names
+
+Jobs have a `name` that is unique to them in their `namespace` by construction.
+
+The Namespace is the root of the naming hierarchy. The job name is constructed to identify the job within that namespace.
+
+Example:
+ - Airflow:
+   - Namespace: the namespace is assigned to the airflow instance. Ex: airflow-staging, airflow-prod
+   - Job: each task in a DAG is a job. name: {dag name}.{task name}
+ - Spark:
+   - Namespace: the namespace is provided as a configuration parameter as in airflow. If there's a parent job, we use the same namespace, otherwise it is provided by configuration.
```

Comment:
I'd move the line about the parent job namespace to the `Parent job run` section

Reply:
Also, what happens if a client reports a job with a namespace and a parent run id, but the namespace doesn't match the parent run's namespace?

Reply:
In that case, the consumer of the event takes the namespace into account in the fully qualified path, using the same namespace allows for a shorter (and more readable) one.